Packs/PAN-OS/Integrations/Panorama/Panorama.py

```diff
@@ -3458,50 +3459,185 @@ def panorama_delete_url_filter_command(url_filter_name: str):
+    ...             'subkey1': 'value1'
+    ...         },
+    ...         {
+    ...             '@xml_attr': 'foo',
+    ...             'random': 'value2'
+    ...         }
+    ...     ]
+    ... }
+    >>> xml_get(xml_dict, 'key', subkeys=('subkey1', 'subkey2'))
+    ['value1', 'value2']
+
+    Args:
+        xml_dict (dict | Any): _description_
+        key (Any): _description_
+        subkeys (tuple, optional): _description_. Defaults to ('member', '#text', '@name').
+
+    Returns:
+        _type_ (str, list, None): _description_
+    """
+
+    if type(xml_dict) is not dict:
+        return xml_dict             # type: ignore[return-value]
+
+    value = xml_dict.get(key)
+
+    if type(value) is dict:
+        return next(
+            (xml_get(value, sub, subkeys)
+             for sub in subkeys
+             if sub in value),
+            None
+        ) or next(
+            (xml_get(value, k)
+             for k in value.keys()
+             if not str(k).startswith('@')),
+            None
+        )
+    if type(value) is list:
+        return [
+            xml_get({0: item}, 0, subkeys)
+            for item in value
+        ]
+
+    return value
+
+
 def prettify_rule(rule: dict):
-    pretty_rule = {
-        'Name': rule['@name'],
-        'Action': rule['action']
-    }
-    if DEVICE_GROUP:
-        pretty_rule['DeviceGroup'] = DEVICE_GROUP
-    if '@loc' in rule:
-        pretty_rule['Location'] = rule['@loc']
-    if isinstance(rule.get('category'), dict) and 'member' in rule['category']:
-        pretty_rule['CustomUrlCategory'] = rule['category']['member']
-    if isinstance(rule.get('application'), dict) and 'member' in rule['application']:
-        pretty_rule['Application'] = rule['application']['member']
-    if isinstance(rule.get('destination'), dict) and 'member' in rule['destination']:
-        pretty_rule['Destination'] = rule['destination']['member']
-    if isinstance(rule.get('from'), dict) and 'member' in rule['from']:
-        pretty_rule['From'] = rule['from']['member']
-    if isinstance(rule.get('service'), dict) and 'member' in rule['service']:
-        pretty_rule['Service'] = rule['service']['member']
-    if isinstance(rule.get('to'), dict) and 'member' in rule['to']:
-        pretty_rule['To'] = rule['to']['member']
-    if isinstance(rule.get('source'), dict) and 'member' in rule['source']:
-        pretty_rule['Source'] = rule['source']['member']
-    if isinstance(rule.get('tag'), dict) and 'member' in rule['tag']:
-        pretty_rule['Tags'] = rule['tag']['member']
-    if isinstance(rule.get('log-setting'), dict) and '#text' in rule['log-setting']:
-        pretty_rule['LogForwardingProfile'] = rule['log-setting']['#text']
-    if disabled := rule.get('disabled'):
-        pretty_rule['Disabled'] = disabled
 
-    return pretty_rule
+    rule_get = partial(xml_get, rule)
+
+    tick_box: Dict[Any, str] = {
+        'yes': '✅',
+        'no': '⬜',
+        None: '⬜',
+    }
```

Comment:
what is this?

Reply:
if we arent using these kinds of checkmarks in the rest of the integration, lets not do it here

Reply:
et tu Judah? then fall tick_box